### ISSUE TITLE
Vision Security Door Sensors state sometimes not correct

### ISSUE THREAD

**Issue by dawiinci** [AlCalzone/node-zwave-js#4468]
I have multiple doors sensor that have never failed like this when using a different platform. Since I switched to zwavejs2mqtt I get false states on all sensors from time to time.

Interestingly if I refresh the values in the interface, it will report the correct state at wakeup.

There are no configuration settings under configure devices, so maybe there is some setting missing? Like the sensor doesn't repeat if the controller doesn't receive the signal? Is this device fully supported?

I am using Home Assistant Core 2022.4.3 without MQTT. 
zwave.me UZB Stick

```
{
  "id": 25,
  "name": "Door 1",
  "loc": "",
  "values": [
    {
      "id": "25-32-0-currentValue",
      "nodeId": 25,
      "commandClass": 32,
      "commandClassName": "Basic",
      "endpoint": 0,
      "property": "currentValue",
      "propertyName": "currentValue",
      "type": "number",
      "readable": true,
      "writeable": false,
      "label": "Current value",
      "stateless": false,
      "commandClassVersion": 0,
      "min": 0,
      "max": 99,
      "list": false,
      "value": 0,
      "isCurrentValue": true,
      "targetValue": "32-0-targetValue",
      "lastUpdate": 1649873732184,
      "newValue": 0
    },
    {
      "id": "25-32-0-targetValue",
      "nodeId": 25,
      "commandClass": 32,
      "commandClassName": "Basic",
      "endpoint": 0,
      "property": "targetValue",
      "propertyName": "targetValue",
      "type": "number",
      "readable": true,
      "writeable": true,
      "label": "Target value",
      "stateless": false,
      "commandClassVersion": 0,
      "min": 0,
      "max": 99,
      "list": false,
      "lastUpdate": 1649873732184
    },
    {
      "id": "25-48-0-Any",
      "nodeId": 25,
      "commandClass": 48,
      "commandClassName": "Binary Sensor",
      "endpoint": 0,
      "property": "Any",
      "propertyName": "Any",
      "type": "boolean",
      "readable": true,
      "writeable": false,
      "label": "Any",
      "ccSpecific": {
        "sensorType": 255
      },
      "stateless": false,
      "commandClassVersion": 0,
      "list": false,
      "value": false,
      "lastUpdate": 1649873732184,
      "newValue": false
    },
    {
      "id": "25-113-0-Home Security-Sensor status",
      "nodeId": 25,
      "commandClass": 113,
      "commandClassName": "Notification",
      "endpoint": 0,
      "property": "Home Security",
      "propertyName": "Home Security",
      "propertyKey": "Sensor status",
      "propertyKeyName": "Sensor status",
      "type": "number",
      "readable": true,
      "writeable": false,
      "label": "Sensor status",
      "ccSpecific": {
        "notificationType": 7
      },
      "stateless": false,
      "commandClassVersion": 0,
      "min": 0,
      "max": 255,
      "list": true,
      "states": [
        {
          "text": "idle",
          "value": 0
        },
        {
          "text": "Intrusion",
          "value": 2
        }
      ],
      "value": 2,
      "lastUpdate": 1649873732184,
      "newValue": 2
    },
    {
      "id": "25-113-0-Home Security-Cover status",
      "nodeId": 25,
      "commandClass": 113,
      "commandClassName": "Notification",
      "endpoint": 0,
      "property": "Home Security",
      "propertyName": "Home Security",
      "propertyKey": "Cover status",
      "propertyKeyName": "Cover status",
      "type": "number",
      "readable": true,
      "writeable": false,
      "label": "Cover status",
      "ccSpecific": {
        "notificationType": 7
      },
      "stateless": false,
      "commandClassVersion": 0,
      "min": 0,
      "max": 255,
      "list": true,
      "states": [
        {
          "text": "idle",
          "value": 0
        },
        {
          "text": "Tampering, product cover removed",
          "value": 3
        }
      ],
      "value": 3,
      "lastUpdate": 1649873732184,
      "newValue": 3
    },
    {
      "id": "25-114-0-manufacturerId",
      "nodeId": 25,
      "commandClass": 114,
      "commandClassName": "Manufacturer Specific",
      "endpoint": 0,
      "property": "manufacturerId",
      "propertyName": "manufacturerId",
      "type": "number",
      "readable": true,
      "writeable": false,
      "label": "Manufacturer ID",
      "stateless": false,
      "commandClassVersion": 0,
      "min": 0,
      "max": 65535,
      "list": false,
      "value": 265,
      "lastUpdate": 1649873732184,
      "newValue": 265
    },
    {
      "id": "25-114-0-productType",
      "nodeId": 25,
      "commandClass": 114,
      "commandClassName": "Manufacturer Specific",
      "endpoint": 0,
      "property": "productType",
      "propertyName": "productType",
      "type": "number",
      "readable": true,
      "writeable": false,
      "label": "Product type",
      "stateless": false,
      "commandClassVersion": 0,
      "min": 0,
      "max": 65535,
      "list": false,
      "value": 8193,
      "lastUpdate": 1649873732184,
      "newValue": 8193
    },
    {
      "id": "25-114-0-productId",
      "nodeId": 25,
      "commandClass": 114,
      "commandClassName": "Manufacturer Specific",
      "endpoint": 0,
      "property": "productId",
      "propertyName": "productId",
      "type": "number",
      "readable": true,
      "writeable": false,
      "label": "Product ID",
      "stateless": false,
      "commandClassVersion": 0,
      "min": 0,
      "max": 65535,
      "list": false,
      "value": 258,
      "lastUpdate": 1649873732184,
      "newValue": 258
    },
    {
      "id": "25-128-0-level",
      "nodeId": 25,
      "commandClass": 128,
      "commandClassName": "Battery",
      "endpoint": 0,
      "property": "level",
      "propertyName": "level",
      "type": "number",
      "readable": true,
      "writeable": false,
      "label": "Battery level",
      "stateless": false,
      "commandClassVersion": 0,
      "min": 0,
      "max": 100,
      "unit": "%",
      "list": false,
      "value": 66,
      "lastUpdate": 1649873732185,
      "newValue": 66
    },
    {
      "id": "25-128-0-isLow",
      "nodeId": 25,
      "commandClass": 128,
      "commandClassName": "Battery",
      "endpoint": 0,
      "property": "isLow",
      "propertyName": "isLow",
      "type": "boolean",
      "readable": true,
      "writeable": false,
      "label": "Low battery level",
      "stateless": false,
      "commandClassVersion": 0,
      "list": false,
      "value": false,
      "lastUpdate": 1649873732185,
      "newValue": false
    },
    {
      "id": "25-132-0-wakeUpInterval",
      "nodeId": 25,
      "commandClass": 132,
      "commandClassName": "Wake Up",
      "endpoint": 0,
      "property": "wakeUpInterval",
      "propertyName": "wakeUpInterval",
      "type": "number",
      "readable": false,
      "writeable": true,
      "label": "Wake Up interval",
      "default": 3600,
      "stateless": false,
      "commandClassVersion": 1,
      "min": 600,
      "max": 604800,
      "step": 200,
      "list": false,
      "value": 43200,
      "lastUpdate": 1649873732185,
      "newValue": 43200
    },
    {
      "id": "25-132-0-controllerNodeId",
      "nodeId": 25,
      "commandClass": 132,
      "commandClassName": "Wake Up",
      "endpoint": 0,
      "property": "controllerNodeId",
      "propertyName": "controllerNodeId",
      "type": "any",
      "readable": true,
      "writeable": false,
      "label": "Node ID of the controller",
      "stateless": false,
      "commandClassVersion": 1,
      "list": false,
      "value": 1,
      "lastUpdate": 1649873732185,
      "newValue": 1
    },
    {
      "id": "25-134-0-libraryType",
      "nodeId": 25,
      "commandClass": 134,
      "commandClassName": "Version",
      "endpoint": 0,
      "property": "libraryType",
      "propertyName": "libraryType",
      "type": "number",
      "readable": true,
      "writeable": false,
      "label": "Library type",
      "stateless": false,
      "commandClassVersion": 0,
      "list": true,
      "states": [
        {
          "text": "Unknown",
          "value": 0
        },
        {
          "text": "Static Controller",
          "value": 1
        },
        {
          "text": "Controller",
          "value": 2
        },
        {
          "text": "Enhanced Slave",
          "value": 3
        },
        {
          "text": "Slave",
          "value": 4
        },
        {
          "text": "Installer",
          "value": 5
        },
        {
          "text": "Routing Slave",
          "value": 6
        },
        {
          "text": "Bridge Controller",
          "value": 7
        },
        {
          "text": "Device under Test",
          "value": 8
        },
        {
          "text": "N/A",
          "value": 9
        },
        {
          "text": "AV Remote",
          "value": 10
        },
        {
          "text": "AV Device",
          "value": 11
        }
      ],
      "value": 6,
      "lastUpdate": 1649873732185,
      "newValue": 6
    },
    {
      "id": "25-134-0-protocolVersion",
      "nodeId": 25,
      "commandClass": 134,
      "commandClassName": "Version",
      "endpoint": 0,
      "property": "protocolVersion",
      "propertyName": "protocolVersion",
      "type": "string",
      "readable": true,
      "writeable": false,
      "label": "Z-Wave protocol version",
      "stateless": false,
      "commandClassVersion": 0,
      "list": false,
      "value": "3.52",
      "lastUpdate": 1649873732185,
      "newValue": "3.52"
    },
    {
      "id": "25-134-0-firmwareVersions",
      "nodeId": 25,
      "commandClass": 134,
      "commandClassName": "Version",
      "endpoint": 0,
      "property": "firmwareVersions",
      "propertyName": "firmwareVersions",
      "type": "string[]",
      "readable": true,
      "writeable": false,
      "label": "Z-Wave chip firmware versions",
      "stateless": false,
      "commandClassVersion": 0,
      "list": false,
      "value": [
        "4.84"
      ],
      "lastUpdate": 1649873732185,
      "newValue": [
        "4.84"
      ]
    }
  ],
  "groups": [
    {
      "text": "Control Command",
      "endpoint": 0,
      "value": 1,
      "maxNodes": 5,
      "isLifeline": true,
      "multiChannel": false
    }
  ],
  "neighbors": [],
  "ready": true,
  "available": true,
  "hassDevices": {},
  "failed": false,
  "inited": true,
  "hexId": "0x0109-0x2001-0x0102",
  "dbLink": "https://devices.zwave-js.io/?jumpTo=0x0109:0x2001:0x0102:4.84",
  "manufacturerId": 265,
  "productId": 258,
  "productType": 8193,
  "deviceConfig": {
    "filename": "/Users/server/zwavejs2mqtt/node_modules/@zwave-js/config/config/devices/0x0109/zd2102.json",
    "isEmbedded": true,
    "manufacturer": "Vision Security",
    "manufacturerId": 265,
    "label": "ZD2102",
    "description": "Door Window Sensor",
    "devices": [
      {
        "productType": 8193,
        "productId": 257
      },
      {
        "productType": 8193,
        "productId": 258
      },
      {
        "productType": 8193,
        "productId": 260
      }
    ],
    "firmwareVersion": {
      "min": "0.0",
      "max": "255.255"
    },
    "associations": {},
    "compat": {
      "valueIdRegex": {},
      "enableBasicSetMapping": true
    }
  },
  "productLabel": "ZD2102",
  "productDescription": "Door Window Sensor",
  "manufacturer": "Vision Security",
  "firmwareVersion": "4.84",
  "protocolVersion": 3,
  "nodeType": 1,
  "endpointsCount": 0,
  "endpointIndizes": [],
  "isSecure": false,
  "security": "None",
  "supportsSecurity": false,
  "supportsBeaming": true,
  "isControllerNode": false,
  "isListening": false,
  "isFrequentListening": false,
  "isRouting": true,
  "keepAwake": false,
  "maxDataRate": 40000,
  "deviceClass": {
    "basic": 4,
    "generic": 32,
    "specific": 1
  },
  "deviceId": "265-258-8193",
  "status": "Asleep",
  "interviewStage": "Complete",
  "minBatteryLevel": 66,
  "batteryLevels": [
    66
  ],
  "statistics": {
    "commandsTX": 1,
    "commandsRX": 1,
    "commandsDroppedRX": 0,
    "commandsDroppedTX": 0,
    "timeoutResponse": 0,
    "rtt": 30.9,
    "rssi": -75,
    "lwr": {
      "protocolDataRate": 2,
      "repeaters": [],
      "rssi": -75,
      "repeaterRSSI": []
    }
  },
  "lastActive": 1649902912688,
  "_name": "Door 1"
}
```

**Comment by robertsLando** [comment]
@dawiinci Zwavejs2mqtt reports what the zwavejs driver reports, driver reports what the device is reporting so here the error could be on device level or driver level. IMO it's a device issue and it needs some compatibility fix. Please upload driver logs

cc @AlCalzone

**Comment by dawiinci** [comment]
I will, but what do you mean by driver log?

**Comment by robertsLando** [comment]
https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs

**Comment by dawiinci** [comment]
Here it is:
[zwavejs.log](https://github.com/zwave-js/node-zwave-js/files/8487844/zwavejs.log)

**Comment by AlCalzone** [comment]
> Interestingly if I refresh the values in the interface, it will report the correct state at wakeup.

Yeah, that's because you instructed zwave-js to ask the device for all its values.

There is no communication at all with the device in your log. In order to troubleshoot, the log needs to run long enough so it contains the situation where the sensor status becomes incorrect.

**Comment by dawiinci** [comment]
The state was frozen at 'on' from 15:55 until 18:08.

The time of the log is - 2 hours.

I have deactivated the manipulation entity in HA. But that shouldn't make a difference.

It is Node 025 (see first and second opening of door - the first one stayed open for HA):

```
2022-04-16T13:55:35.971Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[BasicCCSet]
                                      target value: 255
2022-04-16T13:55:35.972Z CNTRLR   [Node 025] treating BasicCC::Set as a report
2022-04-16T13:55:35.972Z CNTRLR   [Node 025] [~] [Binary Sensor] Any: false => true                 [Endpoint 0]
2022-04-16T13:55:35.990Z SERIAL « 0x0112000400190a710507ff00ff07020000ae0022                          (20 bytes)
2022-04-16T13:55:35.991Z CNTRLR   [Node 025] [~] [Notification] notificationMode: "pull" [Endpoint 0] [internal]
                                   => "pull"
2022-04-16T13:55:35.991Z SERIAL » [ACK]                                                                   (0x06)
2022-04-16T13:55:36.002Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[NotificationCCReport]
                                      notification type:   Home Security
                                      notification status: 255
                                      notification state:  Intrusion
2022-04-16T13:55:36.003Z CNTRLR   [Node 025] [~] [Notification] Home Security[Sensor status]: 2 =>  [Endpoint 0]
                                  2
2022-04-16T13:55:36.015Z SERIAL » 0x010e00137e07600d00022501ff254b41                                  (16 bytes)
2022-04-16T13:55:36.015Z DRIVER » [Node 126] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      75
                                  └─[MultiChannelCCCommandEncapsulation]
                                    │ source:      0
                                    │ destination: 2
                                    └─[BinarySwitchCCSet]
                                        target value: true
2022-04-16T13:55:36.017Z SERIAL « [ACK]                                                                   (0x06)
2022-04-16T13:55:36.023Z SERIAL « 0x0104011301e8                                                       (6 bytes)
2022-04-16T13:55:36.023Z SERIAL » [ACK]                                                                   (0x06)
2022-04-16T13:55:36.023Z DRIVER « [RES] [SendData]
                                    was sent: true
2022-04-16T13:55:36.065Z SERIAL « 0x011800134b00000501c37f7f7f7f000003490000000301000030              (26 bytes)
2022-04-16T13:55:36.066Z SERIAL » [ACK]                                                                   (0x06)
2022-04-16T13:55:36.066Z DRIVER « [REQ] [SendData]
                                    callback id:            75
                                    transmit status:        OK, took 50 ms
                                    repeater node IDs:      73
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 100 kbit/s
                                    ACK RSSI:               -61 dBm
                                    ACK RSSI on repeaters:  N/A
                                    ACK channel no.:        0
                                    TX channel no.:         0
...

2022-04-16T13:55:40.028Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[BasicCCSet]
                                      target value: 0
2022-04-16T13:55:40.039Z SERIAL « 0x0112000400190a7105070000ff07020000b000c3                          (20 bytes)
2022-04-16T13:55:40.040Z CNTRLR   [Node 025] [~] [Notification] notificationMode: "pull" [Endpoint 0] [internal]
                                   => "pull"
2022-04-16T13:55:40.040Z SERIAL » [ACK]                                                                   (0x06)
2022-04-16T13:55:40.040Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[NotificationCCReport]
                                      notification type:   Home Security
                                      notification status: 255
                                      notification state:  Intrusion
2022-04-16T13:55:40.153Z SERIAL « 0x0112000400190a7105070000ff07020000b000c3                          (20 bytes)
2022-04-16T13:55:40.154Z CNTRLR   [Node 025] [~] [Notification] notificationMode: "pull" [Endpoint 0] [internal]
                                   => "pull"
2022-04-16T13:55:40.154Z SERIAL » [ACK]                                                                   (0x06)
2022-04-16T13:55:40.154Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[NotificationCCReport]
                                      notification type:   Home Security
                                      notification status: 255
                                      notification state:  Intrusion
```

```
2022-04-16T16:08:44.697Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[BasicCCSet]
                                      target value: 255
2022-04-16T16:08:44.698Z CNTRLR   [Node 025] treating BasicCC::Set as a report
2022-04-16T16:08:44.699Z CNTRLR   [Node 025] [~] [Binary Sensor] Any: true => true                  [Endpoint 0]
2022-04-16T16:08:44.715Z SERIAL « 0x0112000400190a710507ff00ff07020000ae0022                          (20 bytes)
2022-04-16T16:08:44.716Z CNTRLR   [Node 025] [~] [Notification] notificationMode: "pull" [Endpoint 0] [internal]
                                   => "pull"
2022-04-16T16:08:44.717Z SERIAL » [ACK]                                                                   (0x06)
2022-04-16T16:08:44.717Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[NotificationCCReport]
                                      notification type:   Home Security
                                      notification status: 255
                                      notification state:  Intrusion
2022-04-16T16:08:44.719Z CNTRLR   [Node 025] [~] [Notification] Home Security[Sensor status]: 2 =>  [Endpoint 0]
                                  2

2022-04-16T16:08:48.338Z SERIAL « 0x010b0004001903200100b0007b                                        (13 bytes)
2022-04-16T16:08:48.339Z SERIAL » [ACK]                                                                   (0x06)
2022-04-16T16:08:48.339Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[BasicCCSet]
                                      target value: 0
2022-04-16T16:08:48.340Z CNTRLR   [Node 025] treating BasicCC::Set as a report
2022-04-16T16:08:48.340Z CNTRLR   [Node 025] [~] [Binary Sensor] Any: true => false                 [Endpoint 0]
2022-04-16T16:08:48.358Z SERIAL « 0x0112000400190a7105070000ff07020000b100c2                          (20 bytes)
2022-04-16T16:08:48.358Z CNTRLR   [Node 025] [~] [Notification] notificationMode: "pull" [Endpoint 0] [internal]
                                   => "pull"
2022-04-16T16:08:48.358Z SERIAL » [ACK]                                                                   (0x06)
2022-04-16T16:08:48.375Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[NotificationCCReport]
                                      notification type:   Home Security
                                      notification status: 255
                                      notification state:  Intrusion
2022-04-16T16:08:48.377Z CNTRLR   [Node 025] [~] [Notification] Home Security[Sensor status]: 2 =>  [Endpoint 0]
                                  2
2022-04-16T16:09:03.303Z SERIAL « 0x010f0004001d07600d02022503ffbc00e6                                (17 bytes)


```

**Comment by dawiinci** [comment]
The state seems to have been sent correctly. However HA, did not reflect this, right?

**Comment by AlCalzone** [comment]
Is there anything in the part where you put `...` or did you filter out some more? It looks like the device uses `BasicCCSet` with 255 and 0 to indicate the sensor status
```
022-04-16T16:08:48.339Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[BasicCCSet]
                                      target value: 0
2022-04-16T16:08:48.340Z CNTRLR   [Node 025] treating BasicCC::Set as a report
2022-04-16T16:08:48.340Z CNTRLR   [Node 025] [~] [Binary Sensor] Any: true => false                 [Endpoint 0]
```
but I'm not seeing the mapping at `2022-04-16T13:55:40.028Z`:
```
2022-04-16T13:55:40.028Z DRIVER « [Node 025] [REQ] [ApplicationCommand]
                                  └─[BasicCCSet]
                                      target value: 0
```

It also sends `NotificationCCReport`s, but those are always "active" and never seem to be reset to idle.

**Comment by dawiinci** [comment]
Please have a look at the raw file:
[zwavejs_2022-04-16.log](https://github.com/zwave-js/node-zwave-js/files/8514886/zwavejs_2022-04-16.log)

**Comment by AlCalzone** [comment]
Hmm in that log I see `treating BasicCC::Set as a report` under every incoming `BasicCCSet` command. Some of these are kept as is and `currentValue` gets updated. Some are also mapped to the `Binary Sensor` value, depending on the device class: `[Node 025] treating BasicCC::Set as a report` and some of the devices also send a `Binary Sensor Report` prior to the Basic CC Set, which is why the Basic CC Set doesn't always cause the binary sensor to be updated (because it already did).

I can't find any updates in the log that aren't reflected as a state change.

**Comment by AlCalzone** [comment]
Upon further investigation, this is probably a duplicate of https://github.com/zwave-js/node-zwave-js/issues/4467. Please check if this is fixed for you in v9.0.3, if not feel free to reopen.
